src/Gui/ToolBarManager.cpp

```diff
@@ -36,11 +36,11 @@
 
 using namespace Gui;
 
-ToolBarItem::ToolBarItem()
+ToolBarItem::ToolBarItem() : forceHide(false)
 {
 }
 
-ToolBarItem::ToolBarItem(ToolBarItem* item)
+ToolBarItem::ToolBarItem(ToolBarItem* item, bool forcehide) : forceHide(forcehide)

```

Comment:
As a general rule we should prefer not to introduce these trailing boolean arguments: it creates a problem called "boolean blindness." Basically, we will have some function call in the code with a random boolean appended to it. What does that boolean mean? No one knows! So you can instead structure your code to force the calling code to be a little more verbose by introducing an enumeration. In your case, something like (untested):
```
enum class HideStyle {
    ALWAYS_VISIBLE,
    AUTO_HIDE // Auto-hide when all elements are disabled
};
...
ToolBarItem::ToolBarItem(ToolBarItem* item, HideStyle forcehide) : forceHide(forcehide)
```
so that a call looks like:
```
auto toolBarItem = ToolBarItem(otherItem, ToolBarItem::HideStyle::AUTO_HIDE);
```
There's no mistaking what that final argument is doing now (well, you have to know what AUTO_HIDE means, but that's a separate problem 😄).